### ISSUE TITLE
Groups invitation system

### ISSUE THREAD

**Issue by mahidharc** [WhisperSystems/Signal-Android#1103]
A groups invitation system will be a very welcome addition to the existing features. Basically, if someone adds me to a group, I would like an invite asking me if I want to accept or deny joining that group. All invitations can be collected in a separate screen where they can wait for my confirmations. I know I can always leave a group if I don't want to be on it, but this way, I have control about which groups I join right from the beginning, instead of being dumped into it.

Thanks for the awesome app. :)

**Comment by Invictaz** [comment]
@tinloaf 

This is not true. It's in fact a big problem in Whatsapp in harrasment situations where you can be automatically added to groups with bullies. Telegram has the option that you have to say YES to every group you are added to. If you say NO you will not be added.

Whatsapp support team suggests that if you block the admin(s) of the group they cannot add you anymore. However this does not work:

-Group admins can get a new phone number and add you again
-There can be over 50 admins in a group in total 256 users in a group.

Only solution left is to change your mobile phone number. Which is wrong, since you are the victim of harrasment. 

So TL:DR; please add an option to revoke invitations to a group in order to avoid being added to groups you don't want to get into.

**Comment by Trolldemorted** [comment]
@Invictaz bullies can get a new phone number and harrass you completely without groups. The harassment issue is not really group related.

**Comment by mahidharc** [comment]
Sorry about abandoning this thread after my initial comment. My original intention here was not exactly about harassment, but rather just a matter of simple convenience. I just didn't want to deal with exiting a group in a semi awkward way.

But now, I just don't have this use case anymore. So I'll close this issue for now, since it's just been open for such a long time. I think that a harassment related issue can be dealt by blocking and/or reporting the person. But if anyone feels otherwise, or have some other use case for this feature, maybe we can have this issue reopened.

Thanks for your time.

**Comment by Invictaz** [comment]
Yes harnassment can happen without Signal but such a feature is also convenient if you get invited into a lot of groups.
